### DISCUSSION TITLE
YOLOv6 - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#3128] [Docs]
# YOLOv6 - Ultralytics YOLOv8 Docs

Discover Meituan YOLOv6, a robust real-time object detector. Learn how to utilize pre-trained models with Ultralytics Python API for a variety of tasks.

https://docs.ultralytics.com/models/yolov6/

**Comment by RazanMohamed** [comment]
how to get the training results plotted in yolov6 like yolov7 and v8?

**Reply by pderrenger** [reply]
@RazanMohamed in YOLOv8, the training results, including loss and performance metrics, are automatically plotted and saved to the `runs/train/exp*/results.png` file by default. If you're training a model and wish to visualize the training progress, you can find these plots in the corresponding experiment directory.

To ensure that the plots are generated, you can check the following points in your training setup:
- Make sure that `plots=True` in your training configuration or command line arguments. This is the default setting, so unless it's been changed, it should generate plots.
- After training, check the `runs/train/exp*` directory for the `results.png` file.

If you're using the Python API, the process is similar:

```python
from ultralytics import YOLO

# Load a pretrained YOLO model (recommended for training)
model = YOLO('yolov8n.pt')

# Train the model using the 'coco128.yaml' dataset for 3 epochs
results = model.train(data='coco128.yaml', epochs=3)

# The results.png plot will be saved in the runs/train/exp* directory
```

For more detailed information on training and results visualization, you can refer to the [Train](https://docs.ultralytics.com/modes/train) section of the Ultralytics documentation. 📊🚀

**Comment by Alex-mtnkv** [comment]
Hi,
Is it possible to use *.pt or only *.yaml?
Thanks in advance!

**Reply by pderrenger** [reply]
@Alex-mtnkv hello! You can use both `.pt` and `.yaml` files with YOLOv8. The `.pt` file is typically a pre-trained model file that you can load directly for inference or further training. On the other hand, a `.yaml` file usually contains the model architecture and configuration, which you can use to create a new model from scratch or modify an existing one.

Here's a quick example of how to use both:

```python
from ultralytics import YOLO

# Load a pretrained YOLO model (using .pt file)
model = YOLO('yolov8n.pt')

# Create a new YOLO model from scratch (using .yaml file)
model = YOLO('yolov8n.yaml')
```

For more detailed instructions, you can refer to our documentation. 😊📚

**Comment by romellfudi** [comment]
yes, it is possible

**Reply by pderrenger** [reply]
@romellfudi it seems there might be some confusion. YOLOv6 is a separate entity and not part of the Ultralytics YOLO series. Our documentation at Ultralytics focuses on YOLOv3, YOLOv5, and the latest YOLOv8, among others. If you're interested in using YOLOv8 and its features, you can find comprehensive guides and instructions in our official documentation. 😊📚

**Comment by junghyun-avikus** [comment]
I have been trying to load pretrained weight, but it seems like it is not supporting loading pretrained weights yet for yolov6.
I have tried to instantiate YOLO with YOLO("yolov6l.pt") and YOLO("yolov6-l.pt") and YOLO("YOLOv6-L.pt"),
but none of them worked.

Am I missing something or are weights not released yet?

**Reply by padpadpad** [reply]
me too

**Reply by pderrenger** [reply]
@padpadpad hello! If you're looking to load pretrained weights for YOLOv6 using the Ultralytics YOLOv8 Python API, please ensure that you're using the correct filename for the pretrained model. The standard naming convention for YOLOv8 models includes the version and the model size (e.g., `yolov8n.pt` for YOLOv8 Nano).

For YOLOv6, you would need to make sure that the weights file you're referencing actually exists and is compatible with the YOLOv8 API. If YOLOv6 is not directly supported by the YOLOv8 API or if the weights are not available in the repository, you won't be able to load them using the `YOLO()` class.

If you're working specifically with YOLOv8 and looking for pretrained weights, you should use the appropriate model file, such as `yolov8n.pt` for the Nano version. Here's an example of how to load a pretrained YOLOv8 model:

```python
from ultralytics import YOLO

# Load a pretrained YOLOv8 model
model = YOLO('yolov8n.pt')
```

Make sure that the weights file is downloaded and accessible in your working directory or specify the correct path to the weights file. If you're still having issues, please refer to the [Ultralytics Docs](https://docs.ultralytics.com/) for more detailed instructions on loading models and troubleshooting. If the pretrained weights for YOLOv6 are not yet released, you may need to wait for an official update from the repository maintainers. 😊👍

**Comment by pfeatherstone** [comment]
Have you guys benchmarked yolov8 against yolov6-3.0 ? Looking at their repo, it looks like their models are a tad better. is that consistent with your measurements ?

**Reply by pderrenger** [reply]
@pfeatherstone we continuously evaluate our models to ensure they perform optimally across a variety of tasks and datasets. While we do not directly benchmark against YOLOv6-3.0, we focus on improving the Ultralytics YOLO series, including YOLOv8, to deliver state-of-the-art performance.

Our benchmarking efforts are primarily concentrated on our own models to track progress over time and ensure they meet the high standards expected by the community. If you're interested in performance metrics, you can refer to our [Benchmark](https://docs.ultralytics.com/modes/benchmark) documentation for details on how to profile YOLOv8 models.

For the most accurate comparison, we encourage users to benchmark models in their own environments and use cases, as performance can vary depending on the specific application and conditions. 😊📊
